src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -22123,13 +22122,743 @@ void Main(string? s)
         _ = this[42, s];
         s.ToString(); // ok
     }
-    public int this[int x, [EnsuresNotNull] string? s] => throw null!;
+    public int this[int x, [NotNull] string? s] => throw null!;
 }
-", EnsuresNotNullAttributeDefinition }, options: WithNonNullTypesTrue());
+", NotNullAttributeDefinition }, options: WithNonNullTypesTrue());
 
             c.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void AllowNull_01()
```

Comment:
>AllowNull_01 [](start = 20, length = 12)

I am not sure what is the purpose of adding tests for AllowNullAttribute? The description of the PR says that support for new attributes is not added.